flink-runtime/src/main/java/org/apache/flink/runtime/operators/coordination/OperatorEventValve.java

```diff
@@ -96,42 +94,44 @@ public boolean isShut() {
      * (because it gets dropped through a call to {@link #reset()} or {@link #resetForTask(int)},
      * then the returned future till be completed exceptionally.
      */
-    public CompletableFuture<Acknowledge> sendEvent(
-            SerializedValue<OperatorEvent> event, int subtask) {
-        synchronized (lock) {
-            if (!shut) {
-                return eventSender.apply(event, subtask);
-            }
-
-            final List<BlockedEvent> eventsForTask =
-                    blockedEvents.computeIfAbsent(subtask, (key) -> new ArrayList<>());
-            final CompletableFuture<Acknowledge> future = new CompletableFuture<>();
-            eventsForTask.add(new BlockedEvent(event, subtask, future));
-            return future;
+    public void sendEvent(
+            SerializedValue<OperatorEvent> event,
+            int subtask,
+            CompletableFuture<Acknowledge> result) {
```

Comment:
I am not sure we can do this here, because when the valve is shut, the event and future are stored in a list, and the completion sending (and future completion) is separately triggered later when the valve is opened again.

Reply:
Fair enough. This is also not really important.